src/workloads/execution/SSBColumnStoreIndex.yml

```diff
@@ -0,0 +1,542 @@
+          }
+        }
+      },
+      {
+        $lookup: {
+          from: 'supplier',
+          localField: 'l_suppkey',
+          foreignField: 's_suppkey',
+          as: 'supplier'
+        }
+      },
+      { $match: { 'supplier.nation.region.r_name': 'ASIA' } },
+      {
+        $addFields: {
+          supplier: {
+            $first: '$supplier'
+          }
+        }
+      },
+      {
+        $group: {
+          _id: { order_year: { $year: '$order.o_orderdate' }, brand: '$part.p_brand' },
+          revenue: { $sum: { $multiply: ['$l_extendedprice', '$l_discount'] } }
+        }
+      }
+    ]
+
+Q2_3Query: &Q2_3Query
+  Pipeline:
+    [
+      {
+        $match: {
+          'part.p_brand': 'Brand#21',
+          'part.p_type': 'ECONOMY BRUSHED STEEL'
+        }
+      },
+      {
+        $lookup: {
+          from: 'supplier',
+          localField: 'l_suppkey',
+          foreignField: 's_suppkey',
+          as: 'supplier'
+        }
+      },
+      { $match: { 'supplier.nation.region.r_name': 'AMERICA' } },
+      {
+        $addFields: {
+          supplier: {
+            $first: '$supplier'
+          }
+        }
+      },
+      {
+        $group: {
+          _id: {
+            order_year: {
+              $year: '$order.o_orderdate'
+            },
+            brand: '$part.p_brand'
+          },
+          revenue: { $sum: { $multiply: ['$l_extendedprice', '$l_discount'] } }
+        }
+      }
+    ]
+
+Q3Query: &Q3Query
+  Pipeline:
+    [
+      {
+        "$match": {
+          "order.o_orderdate": {
+            "$gte": ISODate("1992-01-01T00:00:00.000Z"),
+            "$lte": ISODate("1997-01-01T00:00:00.000Z")
+          }
+        }
+      },
+      {
+        "$lookup": {
+          "from": "supplier",
+          "localField": "l_suppkey",
+          "foreignField": "s_suppkey",
+          "as": "supplier"
+        }
+      },
+      {"$addFields": {"supplier": {"$first": "$supplier"}}},
+      {"$match": {"supplier.nation.region.r_name": "ASIA"}},
+      {
+        "$lookup": {
+          "from": "customer",
+          "localField": "order.o_custkey",
+          "foreignField": "c_custkey",
+          "as": "customer"
+        }
+      },
+      {"$addFields": {"customer": {"$first": "$customer"}}},
+      {
+        "$group": {
+          "_id": {
+            "order_year": {"$year": "$order.o_orderdate"},
+            "c_nation": "$customer.nation.n_name",
+            "s_nation": "$supplier.nation.n_name"
+          },
+          "revenue": {"$sum": {"$multiply": ["$l_extendedprice", "$l_discount"]}}
+        }
+      },
+      {
+        "$addFields": {
+          "order_year": "$_id.order_year",
+          "customer_nation": "$_id.c_nation",
+          "supplier_nation": "$_id.s_nation"
+        }
+      },
+      {"$sort": {"order_year": 1, "revenue": -1}}
+    ]
+
+# Q4 queries are (currently) limited to 10000 documents because they take many hours to finish.
+Q4_1Query: &Q4_1Query
```

Comment:
Right, the multiple lookups drag us down here, which is why we had to limit it to 10000 documents for now. Denormalizing to avoid them is an interesting idea though.